src/shared/codelens/pythonCodeLensProvider.ts

```diff
@@ -0,0 +1,122 @@
+/*!
+ * Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+'use strict'
+
+import * as path from 'path'
+import * as vscode from 'vscode'
+
+import { DebugConfiguration } from '../../lambda/local/debugConfiguration'
+import { LambdaHandlerCandidate } from '../lambdaHandlerSearch'
+import { getLogger } from '../logger'
+import { DefaultSamCliProcessInvoker, DefaultSamCliTaskInvoker, } from '../sam/cli/samCliInvoker'
+import { getDebugPort } from '../utilities/vsCodeUtils'
+
+import {
+    CodeLensProviderParams,
+    getInvokeCmdKey,
+    makeCodeLenses
+} from './codeLensUtils'
+import { LambdaLocalInvokeArguments, LocalLambdaRunner } from './localLambdaRunner'
+
+export const PYTHON_LANGUAGE = 'python'
+export const PYTHON_ALLFILES: vscode.DocumentFilter[] = [
+    { language: PYTHON_LANGUAGE }
+]
+
+// TODO: Fix this! Implement a more robust/flexible solution. This is just a basic minimal proof of concept.
+const getSamProjectDirPathForFile = async (filepath: string): Promise<string> => {
+    return path.dirname(filepath)
+}
+
+export const getLambdaHandlerCandidates = async ({ uri }: { uri: vscode.Uri }): Promise<LambdaHandlerCandidate[]> => {
+    const logger = getLogger()
+    const filename = uri.fsPath
+
+    logger.info(`Getting symbols for '${uri.fsPath}'`)
+    const symbols: vscode.DocumentSymbol[] = ( // SymbolInformation has less detail (no children)
+        (await vscode.commands.executeCommand<vscode.DocumentSymbol[]>(
+            'vscode.executeDocumentSymbolProvider',
+            uri
+        )) || []
+    )
+
+    return symbols
+        .filter(sym => sym.kind === vscode.SymbolKind.Function)
+        .map(symbol => {
+            logger.debug(`Found potential handler: '${path.parse(filename).name}.${symbol.name}'`)
+
+            return {
+                filename,
+                handlerName: `${path.parse(filename).name}.${symbol.name}`,
+                range: symbol.range
+            }
+        })
+}
+
+export const initialize = ({
+    configuration,
+    toolkitOutputChannel,
+    processInvoker = new DefaultSamCliProcessInvoker(),
+    taskInvoker = new DefaultSamCliTaskInvoker()
+}: CodeLensProviderParams): void => {
+    const logger = getLogger()
+    vscode.commands.registerCommand(
```

Comment:
Please use the telemetry registerCommand call

Reply:
Merged and refactored to work with recent telemetry changes.
✔️